javascript/mobile_questionnaire.js

```diff
@@ -339,6 +387,171 @@ function textBoxObserver(mutationList, observer) {
     });
 }
 
+function dateTimeObserver(mutationList, observer) {
+    mutationList.forEach((mutation) => {
+
+        switch (mutation.type) {
+            case 'attributes':
+
+                var currentRequiredValue = mutation.target.getAttribute('data-currentinput');
+                var finalRequiredInput = mutation.target.getAttribute('data-finalinput');
+                var button = document.getElementsByClassName('questionnaire submit-button');
+
+                if (!currentRequiredValue && !finalRequiredInput) {
+                    return;
+                }
+
+                if (mutation.target.innerText == '') {
+                    for (var x = 0; x < button.length; x++) {
+                        button[x].disabled = true;
+                    }
+                    return;
+                }
+                // Might need to build a check function to make sure that the date time adheres to certain policies
+
+                if (!requiredInputs.includes(currentRequiredValue)) {
+                    requiredInputs.push(currentRequiredValue); // Only push if it has not been added to the array already.
+                }
+
+                var numberOfRequiredAnswers = 0;
+                for (var x = 0; x < requiredInputs.length; x++) {
+                    // First need to check that all answers before required answer are in array.
+                    // Then set a flag that I can check later.
+                    numberOfRequiredAnswers++;
+                }
+
+                var requiredInput = false;
+                if (requiredInputs.includes(finalRequiredInput)) {
+                    requiredInput = true;
+                }
+
+                var nextButton = document.getElementsByClassName('questionnaire next-button');
+
+                if (requiredInput === true && numberOfRequiredAnswers == finalRequiredInput && typeof(button) != 'undefined') {
+                    for (var x = 0; x < button.length; x++) {
+                        button[x].disabled = false;
+                    }
+                }
+                if (requiredInput === true && numberOfRequiredAnswers == finalRequiredInput && typeof(nextButton) != 'undefined') {
+                    for (var i = 0; i < nextButton.length; i++){
+                        nextButton[i].disabled = false;
+                    }
+                }
+            break;
```

Comment:
No sure if worth it adding default case statements. If you do not see a need for a default handle then this should be ok I guess.

Reply:
shot

Reply:
I remember there being a stylistic thing to the codechecker, moodle does not want you to have a switch without a break statement

Reply:
no need for a default statement, since I am trying to keep it to a single check for the observer